ui/src/components/domutil.ts

```diff
@@ -258,3 +258,30 @@ export function computeTop(fromBottom = false, ctx: PartialPositionContext) {
 
   return top;
 }
+
+/**
+ * Compute the "height" property for `ctx.element`.
+ *
+ * @param fromBottom whether or not the element should be "bottom" or "top" aligned
+ * @param ctx the position context used to make positioning computations
+ */
+export function computeHeight(fromBottom = false, ctx: PartialPositionContext) {
+  let height;
+  const top = computeTop(fromBottom, ctx);
+  const { body } = completePositionContext(ctx);
+  const bodyHeight = body.height;
+
+  /**
+   * Restrict the height when necessary
+   * We assume that if the top of the element is above the middle of the page,
+   * we will have enough space to let it grow.
+   */
+  if (top < bodyHeight / 2) {
+    // leave the default height
+    height = undefined;
+  } else {
+    // avoid the element to overflow the body
+    height = bodyHeight - top;
+  }
```

Comment:
completely!

Reply:
So, I tried with max-height, but it can't work the same way.

Indeed, as super well explained [here](https://stackoverflow.com/a/14263416/19354626), “when you don't specify an explicit height on the parent, then there's no base height for the child's height to be calculated from, so max-height computes to none, allowing the child to be as tall as possible”

 I made a video to try to explain better the issue.

https://user-images.githubusercontent.com/18734475/207436399-e32cc8ce-d438-4d75-888f-f5fc86e5084a.mp4

I think that the only solution left is to pass the max-height value to the child, but I didn’t succeed to apply the style through slot + component, and it's not really clean 😟

Or we can keep the height property 😛